### PR TITLE
[TSI] Add code snippets for hierarchies and types

### PR DESCRIPTION
**Pull request by bikamani**
Adding code snippets for hierarchies and types.

### CODE REVIEW

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 1]
```diff
@@ -40,6 +40,16 @@ internal HierarchiesClient(TimeSeriesHierarchiesRestClient hierarchiesRestClient
         /// </summary>
         /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>The pageable list <see cref="AsyncPageable{TimeSeriesHierarchy}"/> of Time Series hierarchies with the http response.</returns>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleGetAllHierarchies">
+        /// // Get all Time Series hierarchies in the environment
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:45]: extra //

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 2]
```diff
@@ -274,6 +284,33 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        /// {
+        ///     &quot;sampleHierarchyId&quot;
+        /// };
+        ///
+        /// Response&lt;TimeSeriesHierarchyOperationResult[]&gt; getHierarchiesByIdsResult = await client
+        ///             .Hierarchies
+        ///             .GetByIdAsync(tsiHierarchyIds)
+        ///             .ConfigureAwait(false);
+        ///
+        /// // The response of calling the API contains a list of hieararchy or error objects corresponding by position to the input parameter array in the request.
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:299]: extra //

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 3]
```diff
@@ -274,6 +284,33 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        /// // If the error object is set to null, this means the operation was a success.
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:300]: extra //

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 4]
```diff
@@ -377,6 +414,42 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        ///
+        /// var tsiHierarchy = new TimeSeriesHierarchy(tsiHierarchyName, hierarchySource);
+        /// tsiHierarchy.Id = &quot;sampleHierarchyId&quot;;
+        ///
+        /// var timeSeriesHierarchies = new List&lt;TimeSeriesHierarchy&gt;
+        /// {
+        ///     tsiHierarchy
+        /// };
+        ///
+        /// // Create Time Series hierarchies
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:432]: extra //

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 5]
```diff
@@ -377,6 +414,42 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        /// Response&lt;TimeSeriesHierarchyOperationResult[]&gt; createHierarchiesResult = await client
+        ///     .Hierarchies
+        ///     .CreateOrReplaceAsync(timeSeriesHierarchies)
+        ///     .ConfigureAwait(false);
+        ///
+        /// // The response of calling the API contains a list of error objects corresponding by position to the input parameter array in the request.
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:438]: extra //

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 6]
```diff
@@ -377,6 +414,42 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
...
+        /// // If the error object is set to null, this means the operation was a success.
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:439]: extra //

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs [hunk 7]
```diff
@@ -579,6 +652,34 @@ Page<TimeSeriesHierarchy> NextPageFunc(string nextLink, int? pageSizeHint)
         /// <exception cref="ArgumentException">
         /// The exception is thrown when <paramref name="timeSeriesHierarchyIds"/> is empty.
         /// </exception>
+        /// <example>
+        /// <code snippet="Snippet:TimeSeriesInsightsSampleDeleteHierarchiesById">
+        /// // Delete Time Series hierarchies with Ids
```
Comment (vinagesh) [inline sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:657]: Just look for all extra // and remove

**Review by vinagesh — APPROVED** [review]

File: sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs (hunk 1 above)
Comment (bikamani) [reply sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/HierarchiesClient.cs:45]: That's coming from a sample snippet, comments in the snippet will look like this which is fine.

**Pull request merged**